src/basic/terminal-util.c

```diff
@@ -1068,8 +1097,8 @@ int ptsname_namespace(int pty, char **ret) {
         return 0;
 }
 
-int openpt_in_namespace(pid_t pid, int flags) {
-        _cleanup_close_ int pidnsfd = -1, mntnsfd = -1, usernsfd = -1, rootfd = -1;
+int openpt_allocate_in_namespace(pid_t pid, int flags, int *master, char **console) {
```

Comment:
why return the fd as parameter instead of as return value like before?

also, as above, console → ret_slave?